src/BalanceWithCurrency.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Wearesho\Delivery;
+
+/**
+ * Class BalanceWithCurrency
+ * @package Wearesho\Delivery
+ */
+class BalanceWithCurrency extends Balance implements ContainsCurrency
```

Comment:
Why not to implement single class with two properties `amount` and `currency` where currency is optional?

Reply:
I agree, it will be more optimals